sherlockml/clients/auth.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2018 ASI Data Science
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from datetime import datetime, timedelta
+import pytz
+from collections import namedtuple
+
+import requests
+
+
+AccessToken = namedtuple('AccessToken', ['token', 'expires_at'])
+
+
+class HudsonClient(object):
+    """Session with the SherlockML authentication service."""
+
+    def __init__(self, url):
+        self._session = requests.Session()
+        self.url = url
+
+    def get_access_token(self, client_id, client_secret):
```

Comment:
Do we expect the same client to be used for multiple identities? I would have expected the client ID and secret to be instance attributes passed at instantiation.

Reply:
I think having both url and self.url might be confusing. Could either call this variable endpoint or change self.url to , e.g,self.host.

Reply:
@srstevenson - I live having the arguments of the method reflect the parameters of the API request, since the client method is intended to encapsulate those requests in code. Also, I would like to encourage less reliance on state in classes where possible, for testing / maintenance benefits.